### PR TITLE
Jira V3 fetch incidents by id error

### PR DESCRIPTION
**Pull request by anas-yousef**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7135

## Description
When fetching by ID, and the *Issue index to start fetching incidents from* is not found in the Jira instance, the integration returns an error: *Issue does not exist or you do not have permission to see it*. The fetch incidents should be able to handle such a case.
For example, when *Issue index to start fetching incidents from* is equal to 0, and there is no issue with index 0, then the error is returned.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
Great work! 
Added a few comments please review them.

**Comment by ShirleyDenkberg** [comment]
@ShacharKidor Doc review completed.

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 1]
```diff
@@ -3680,7 +3716,7 @@ def update_remote_system_command(client: JiraBaseClient, args: Dict[str, Any], c
                 f'Got the following delta keys {list(delta.keys())} to update JiraV3 Incident {remote_id}'
             )
             # take the val from data as it's the updated value
-            delta = {k: remote_args.data.get(k) for k in delta.keys()}
+            delta = {k: remote_args.data.get(k) for k in delta}
```
[the remark this answers is not in the document]
Comment (anas-yousef) [reply Packs/Jira/Integrations/JiraV3/JiraV3.py:3719]: Yup, it was actually suggested by the [Ruff](https://beta.ruff.rs/docs/rules/in-dict-keys/)

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 2]
```diff
@@ -2998,11 +3013,20 @@ def fetch_incidents(client: JiraBaseClient, issue_field_to_fetch_from: str, fetc
...
     # to be integers (if we want to use the issues' ids in the query, they must be passed on as integers and not strings),
     # we convert the list to hold integer values
     last_fetch_issue_ids: List[int] = convert_list_of_str_to_int(last_run.get('issue_ids', []))
     last_fetch_id = last_run.get('id', id_offset)
+    if last_fetch_id in [0, '0'] and issue_field_to_fetch_from == 'id':
+        # If last_fetch_id is equal to zero, and the user wants to fetch using the issue ID, then we automatically
+        # acquire the smallest issue ID with respect to the query
+        _, smallest_id_offset = get_smallest_id_offset_for_query(client=client, query=fetch_query)
+        if not smallest_id_offset:
+            raise DemistoException('The fetch query configured returned no results, therefore, could not start fetching.')
```
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/Jira/Integrations/JiraV3/JiraV3.py]: yes, it is better :)

**Review by ShacharKidor — APPROVED** [review]

**Pull request merged**